### PR TITLE
ROX-17114: Fix misattribution of Java versions when there are multiple matching lines

### PR DESCRIPTION
**Pull request by connorgorman**
So we don't stop early if we have matched all the possible key lines in the manifest.mf file. This file can contain many entries and this will cause us to attribute the data to a wrong version

### CODE REVIEW

File: pkg/analyzer/java/parse_manifest_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package java
+
+import (
+	"bytes"
+	"fmt"
+	"testing"
+)
+
+const testManifestMF = `
```
Comment (daynewlee) [inline pkg/analyzer/java/parse_manifest_test.go:9]: @RTann 
Maybe we should add this specific example to the Claircore to manifest parsing. What do you think?

File: pkg/analyzer/java/parse_manifest_test.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+Implementation-Title: jakarta.security.auth.message
+Implementation-Vendor: Apache Software Foundation
+Implementation-Version: 3.0
+Specification-Title: Jakarta Authentication SPI for Containers
+Specification-Vendor: Eclipse Foundation
+Specification-Version: 3.0
+`
+
+func TestStopOnceFirstKeyIsFound(t *testing.T) {
+	manifestBytes := bytes.NewBuffer([]byte(testManifestMF))
```
Comment (connorgorman) [inline pkg/analyzer/java/parse_manifest_test.go:42]: Whoops, I'll added assertions here

**Review by daynewlee — APPROVED** [review]
LGTM

File: pkg/analyzer/java/parse_manifest_test.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+Implementation-Title: javax.servlet
+Implementation-Vendor: Apache Software Foundation
+Implementation-Version: 4.0.FR
+Specification-Title: Java API for Servlets
+Specification-Vendor: Sun Microsystems, Inc.
+Specification-Version: 4.0
+`
+
+func TestStopOnceFirstKeyIsFound(t *testing.T) {
+	manifest, err := parseManifestMFFromReader("", bytes.NewBufferString(testTomcat9ManifestMF))
```
Comment (RTann) [inline pkg/analyzer/java/parse_manifest_test.go:153]: can you check `Bundle-Version` too for completeness?

File: pkg/analyzer/java/parse_manifests.go [hunk 4]
```diff
@@ -83,7 +76,7 @@ func parseManifestMF(locationSoFar string, f *zip.File) (parsedManifestMF, error
 			isTemplatedManifestFile = true
 			continue
 		}
-
+		valueFromLine = strings.TrimSpace(valueFromLine)
```
Comment (RTann) [inline pkg/analyzer/java/parse_manifests.go:79]: should we do this before the `if valueFromLine == ""` check?

Comment (connorgorman) [reply pkg/analyzer/java/parse_manifests.go:79]: Ah yeah, good call

**Comment by RTann** [comment]
I will be taking over this PR at this point

**Review by daynewlee — APPROVED** [review]

File: pkg/analyzer/java/parse_manifest_test.go (hunk 1 above)
Comment (RTann) [reply pkg/analyzer/java/parse_manifest_test.go]: ClairCore doesn't have this problem, as they only read the "main" section and also stop at `Name`: https://github.com/quay/claircore/blob/main/java/jar/jar.go#L526

**Pull request merged**